train.py

```diff
@@ -198,6 +201,7 @@ def train_epoch(
                 saved_var[tensor_name].add_(new_grad)
             model.zero_grad()
 
+        gamma += (b/num_microbatches-gamma)/2
```

Comment:
I’ve been thinking about this more. Intuitively, we want to favor a clipping threshold at the beginning of training that has low bias, but higher variance. In this case, it’s ok that we add a higher clipping threshold in exchange for more noise because the low bias in the estimate helps the model converge towards a good initial solution. However, later on, its important that there is little noise added to the parameter estimate (low variance) so that the model can learn finer grained details. 

See discussion by [Andrew et. al](https://arxiv.org/pdf/1905.03871.pdf), for example
<img width="1159" alt="image" src="https://user-images.githubusercontent.com/5036922/165882709-39b35123-2f12-431d-8fdf-6e2810364df6.png">